plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3BlobContainer.java

```diff
@@ -685,41 +700,47 @@ static Tuple<Long, Long> numberOfMultiparts(final long totalSize, final long par
      * the stream and its related metadata.
      * @param s3AsyncClient Async client to be utilized to fetch the object part
      * @param bucketName Name of the S3 bucket
-     * @param blobName Identifier of the blob for which the parts will be fetched
-     * @param partNumber Part number for the blob to be retrieved
+     * @param blobKey Identifier of the blob for which the parts will be fetched
+     * @param partNumber Optional part number for the blob to be retrieved
      * @return A future of {@link InputStreamContainer} containing the stream and stream metadata.
      */
     CompletableFuture<InputStreamContainer> getBlobPartInputStreamContainer(
         S3AsyncClient s3AsyncClient,
         String bucketName,
-        String blobName,
-        int partNumber
+        String blobKey,
+        @Nullable Integer partNumber
     ) {
-        final GetObjectRequest.Builder getObjectRequestBuilder = GetObjectRequest.builder()
-            .bucket(bucketName)
-            .key(blobName)
-            .partNumber(partNumber);
+        final boolean isMultipartObject = partNumber != null;
+        final GetObjectRequest.Builder getObjectRequestBuilder = GetObjectRequest.builder().bucket(bucketName).key(blobKey);
+
+        if (isMultipartObject) {
+            getObjectRequestBuilder.partNumber(partNumber);
+        }
 
         return SocketAccess.doPrivileged(
             () -> s3AsyncClient.getObject(getObjectRequestBuilder.build(), AsyncResponseTransformer.toBlockingInputStream())
-                .thenApply(S3BlobContainer::transformResponseToInputStreamContainer)
+                .thenApply(response -> transformResponseToInputStreamContainer(response, isMultipartObject))
         );
     }
 
     /**
      * Transforms the stream response object from S3 into an {@link InputStreamContainer}
      * @param streamResponse Response stream object from S3
+     * @param isMultipartObject Flag to denote a multipart object response
      * @return {@link InputStreamContainer} containing the stream and stream metadata
      */
     // Package-Private for testing.
-    static InputStreamContainer transformResponseToInputStreamContainer(ResponseInputStream<GetObjectResponse> streamResponse) {
+    static InputStreamContainer transformResponseToInputStreamContainer(
+        ResponseInputStream<GetObjectResponse> streamResponse,
+        boolean isMultipartObject
+    ) {
         final GetObjectResponse getObjectResponse = streamResponse.response();
         final String contentRange = getObjectResponse.contentRange();
         final Long contentLength = getObjectResponse.contentLength();
-        if (contentRange == null || contentLength == null) {
+        if ((isMultipartObject && contentRange == null) || contentLength == null) {
```

Comment:
Is contentRange actually null in the non multipart case? I would expect it always to be present and not to need conditional logic here.

Reply:
It is `null` in the non multipart case. I tried looking up the client contract for S3 - could not locate this specific attribute behavior. But in my tests - it is null in non multipart cases.